### PR TITLE
G3000 enhanced merge

### PR DESCRIPTION
**Pull request by cywang90**
Integrated the G3000 enhanced mod. Due to changes to MapInstrument, I tested G1000 and did not find any issues. I did have to make a minor change to the G1000 folder to make the G1000 MFD compatible with the UI changes I made to the orientation and range display insets. The change was in AS1000_MFD.css and I had to bring the entire file into the WT directory (I tried putting the code into AS1000_MFD_WT.css but couldn't get it to work - I'm guessing because I was trying to set styles that were already set in AS1000_MFD.css).

### CODE REVIEW

**Comment by kaosfere** [comment]
This is awesome, thank you very much!   I've put the rest of the team on the PR so they're aware of it too.   I'm going to make my way through the changes you made to the altered files in our repo and see if I see any issues, but initially this seems great.

Will respond with more as soon as I an.  :D

File: src/workingtitle-vcockpits-instruments/html_ui/Pages/VCockpit/Instruments/Shared/Map/WorkingTitle/MapInstrument.css [hunk 1]
```diff
@@ -38,16 +38,28 @@ map-instrument {
...
     position: absolute;
     bottom: 0%;
     right: 0%;
-    width: 5%;
-    height: calc( 6 * 1vh * var(--bodyHeightScale));
+    width: 10%;
+    height: calc( 4 * 1vh * var(--bodyHeightScale));
     background-color: #1a1d21;
-    color: aquamarine;
-    text-align: center; }
```
Comment (kaosfere) [inline src/workingtitle-vcockpits-instruments/html_ui/Pages/VCockpit/Instruments/Shared/Map/WorkingTitle/MapInstrument.css:50]: @knighty is very happy about this change.   This has drivin him crazy.  :D

File: src/workingtitle-vcockpits-instruments/html_ui/Pages/VCockpit/Instruments/Shared/Map/WorkingTitle/MapInstrument.js [hunk 2]
```diff
@@ -1174,20 +1213,48 @@ class MapInstrument extends ISvgMapRootElement {
...
-    }
+        if (this.orientation == "north" || !this.offsetPlaneInHdgTrk) {
+			this.setNavMapCenter(this.navMap.planeCoordinates);
+			if (this.eBingMode == EBingMode.PLANE) {
+				this.airplaneIconElement.forceCoordinates(this.navMap.centerCoordinates.lat, this.navMap.centerCoordinates.long);
+			}
+		} else {
+			// MOD: if orientation is heading or track up, we want to place the plane 33% from the bottom of the map,
+			// but vector needs to be adjusted for overdraw factor of sqrt(2), which is where the magic number of 382 comes from (500-500/sqrt(2)+0.33*1000/sqrt(2))
+			let target = this.navMap.XYToCoordinatesFromPlaneWithRotation(new Vec2(500, 382));
```
Comment (kaosfere) [inline src/workingtitle-vcockpits-instruments/html_ui/Pages/VCockpit/Instruments/Shared/Map/WorkingTitle/MapInstrument.js:1226]: This is a great detail, it hadn't really hit me that the plane is offset from center when not in north-up mode.

File: src/workingtitle-vcockpits-instruments-navsystems-g3000/html_ui/WorkingTitle/Pages/VCockpit/Instruments/NavSystems/AS3000/MFD/AS3000_MFD.js [hunk 3]
```diff
@@ -0,0 +1,768 @@
...
+        if (_event == "SOFTKEYS_12") {
+            this.acknowledgeInit();
+        }
+    }
+    Update() {
+        super.Update();
+    }
+}
+
+class AS3000_MFD_WindDataDisplay extends HTMLElement {
```
Comment (kaosfere) [inline src/workingtitle-vcockpits-instruments-navsystems-g3000/html_ui/WorkingTitle/Pages/VCockpit/Instruments/NavSystems/AS3000/MFD/AS3000_MFD.js:38]: Can you summarize how this differs from the common wind data element?

File: src/workingtitle-vcockpits-instruments-navsystems-g3000/html_ui/WorkingTitle/Pages/VCockpit/Instruments/NavSystems/AS3000/AS3000_Common.js [hunk 4]
```diff
@@ -0,0 +1,300 @@
+class AS3000_MapElement extends MapInstrumentElement {
```
Comment (kaosfere) [inline src/workingtitle-vcockpits-instruments-navsystems-g3000/html_ui/WorkingTitle/Pages/VCockpit/Instruments/NavSystems/AS3000/AS3000_Common.js:1]: I'd be interested in how much of this we can pull out into a common space and put into Navsystems.js or CommonPFD_MFD.js so that it can be used by any nav system.   I'm sure features will vary between units, but I suspect the logic will be highly similar.

That's not a change requested for this PR, just something I think we should think about down the road.

File: src/workingtitle-vcockpits-instruments-navsystems-g3000/html_ui/WorkingTitle/Pages/VCockpit/Instruments/NavSystems/AS3000/MFD/AS3000_MFD.js (hunk 3 above)
Comment (cywang90) [reply src/workingtitle-vcockpits-instruments-navsystems-g3000/html_ui/WorkingTitle/Pages/VCockpit/Instruments/NavSystems/AS3000/MFD/AS3000_MFD.js:38]: Sure. The graphics are a bit different. Arrow is fatter and shorter. The wind strength display now says "KT" after the number. It also will hide the arrow if wind speed is <1 kt. It will also show NO WIND DATA if the plane's true ground speed is less than 5 knots. I tried to make it as close to the G3000's actual MFD wind vector display as possible. I think I got it pretty close; the only dodgy bit is the no wind data <5 kt thing. I couldn't find a description anywhere of when exactly the G3000 does and does not report wind data. The TBM G3000 manual just says it won't do it when the aircraft is "not moving", which is very vague.

File: src/workingtitle-vcockpits-instruments/html_ui/Pages/VCockpit/Instruments/Shared/Map/WorkingTitle/Svg/SvgRangeCompass.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+class SvgRangeCompass extends SvgMapElement {
```
Comment (kaosfere) [inline src/workingtitle-vcockpits-instruments/html_ui/Pages/VCockpit/Instruments/Shared/Map/WorkingTitle/Svg/SvgRangeCompass.js:1]: Is this the start of an attempt to draw range rings on the map?

File: src/workingtitle-aircraft-tbm930/SimObjects/Airplanes/Asobo_TBM930/model/TBM930_interior.xml [hunk 6]
```diff
@@ -0,0 +1,2228 @@
...
+		<LOD minSize="70" ModelFile="TBM930_interior_LOD01.gltf"/>
+		<LOD minSize="20" ModelFile="TBM930_interior_LOD02.gltf"/>
+		<LOD minSize="10" ModelFile="TBM930_interior_LOD03.gltf"/>
+		<LOD minSize="5" ModelFile="TBM930_interior_LOD04.gltf"/>
+		<LOD minSize="2" ModelFile="TBM930_interior_LOD05.gltf"/>
+	</LODS>
+	<Behaviors>
+		<!-- Includes ###################################### -->
+		<Include ModelBehaviorFile="Asobo\Common.xml"/>
+		<Include ModelBehaviorFile="Asobo\GlassCockpit\AS3000_enhanced.xml"/>
```
Comment (kaosfere) [inline src/workingtitle-aircraft-tbm930/SimObjects/Airplanes/Asobo_TBM930/model/TBM930_interior.xml:17]: I don't think these are going to work any more, those files don't exist in this repo.

File: src/workingtitle-vcockpits-instruments-navsystems-g3000/html_ui/WorkingTitle/Pages/VCockpit/Instruments/NavSystems/AS3000/MFD/AS3000_MFD.js (hunk 3 above)
Comment (kaosfere) [reply src/workingtitle-vcockpits-instruments-navsystems-g3000/html_ui/WorkingTitle/Pages/VCockpit/Instruments/NavSystems/AS3000/MFD/AS3000_MFD.js:38]: Interesting.   The fix we put in CommonPFD_MFD.js for the shared wind bug just checks to see whether the plane is on the ground.    The question of what happens on the takeoff run is interesting.   I just fired up my G1000 trainer, and at least in that I could be doing 200kts on the ground and it still said NO WIND DATA.

Not sure how the 3000 should be have, though.

**Review by kaosfere — COMMENTED** [review]
Overall summary:    i really, really like this.   I'm happy to see you started using some of our settings-persistent gauges, too.  :D 

You mentioned the conflict with the ModelBehaviorDefs... it's a good point.  If we don't separate them, we'll end up shipping "custom" XML files to people who don't have the particular mod that they're needed for and that's not very polite.

Maybe we should just have those start living with the individual avionics mods and we can stop worry about fs-base-aircraft-common.  What do you think about that?

File: src/workingtitle-aircraft-tbm930/SimObjects/Airplanes/Asobo_TBM930/model/TBM930_interior.xml (hunk 6 above)
Comment (cywang90) [reply src/workingtitle-aircraft-tbm930/SimObjects/Airplanes/Asobo_TBM930/model/TBM930_interior.xml:17]: Oh yeah, missed that one. Easy fix to just change the file path.

File: src/workingtitle-vcockpits-instruments/html_ui/Pages/VCockpit/Instruments/Shared/Map/WorkingTitle/Svg/SvgRangeCompass.js (hunk 5 above)
Comment (cywang90) [reply src/workingtitle-vcockpits-instruments/html_ui/Pages/VCockpit/Instruments/Shared/Map/WorkingTitle/Svg/SvgRangeCompass.js:1]: Yes. I forgot that file was in there when I did the mass copy-paste.

File: src/workingtitle-vcockpits-instruments-navsystems-g3000/html_ui/WorkingTitle/Pages/VCockpit/Instruments/NavSystems/AS3000/MFD/AS3000_MFD.js (hunk 3 above)
Comment (cywang90) [reply src/workingtitle-vcockpits-instruments-navsystems-g3000/html_ui/WorkingTitle/Pages/VCockpit/Instruments/NavSystems/AS3000/MFD/AS3000_MFD.js:38]: Yeah, I actually initially also did a ON GROUND check. But I kept reading that one sentence in the manual over and over again that says no wind data when not moving and I figured "well technically you can be moving and still be on the ground..." But I don't know what the right answer is either.

**Comment by cywang90** [comment]
> Overall summary: i really, really like this. I'm happy to see you started using some of our settings-persistent gauges, too. :D
> 
> You mentioned the conflict with the ModelBehaviorDefs... it's a good point. If we don't separate them, we'll end up shipping "custom" XML files to people who don't have the particular mod that they're needed for and that's not very polite.
> 
> Maybe we should just have those start living with the individual avionics mods and we can stop worry about fs-base-aircraft-common. What do you think about that?

I like that idea. It's a clean solve and I can't really think of a downside right now.

File: src/workingtitle-aircraft-tbm930/SimObjects/Airplanes/Asobo_TBM930/model/TBM930_interior.xml (hunk 6 above)
Comment (cywang90) [reply src/workingtitle-aircraft-tbm930/SimObjects/Airplanes/Asobo_TBM930/model/TBM930_interior.xml]: Fixed. af150ce11262a3bb1dec23a07298f3a7aaa56562

File: src/workingtitle-vcockpits-instruments/html_ui/Pages/VCockpit/Instruments/Shared/Map/WorkingTitle/Svg/SvgRangeCompass.js (hunk 5 above)
Comment (kaosfere) [reply src/workingtitle-vcockpits-instruments/html_ui/Pages/VCockpit/Instruments/Shared/Map/WorkingTitle/Svg/SvgRangeCompass.js:1]: Haha, no worries.   It's something that's been requested  of us already, so I don't mind having the stub there.  Hopefully we'll be able to flesh it out sooner rather than later.

**Comment by kaosfere** [comment]
Cool, looks solid.  I'm OK merging this without changing the ModelBehaviorDefs location then doing it across the repo as one change, might help avoid creating any conflicts.   I'm not going to click approve yet, as I'd like to let some of the other folks on the team chime in, but I'm very happy/excited over what I see here.  It gets a 👍  from my perspective.

**Review by kaosfere — APPROVED** [review]
OK, this looks solid to me and the rest of the team seems positive.   Let's give it a go.
